src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -570,3 +570,28 @@ class KeyedAggregator[T, K](aggregator: TypedAggregator[T]) extends TypedAggrega
 
   def copy() = new KeyedAggregator(aggregator.copy())
 }
+
+class DownsampleAggregator(nDivisions: Int, getY: Any => Any) extends TypedAggregator[IndexedSeq[Row]] {
+  require(nDivisions > 0)
+
+  var _state = new DownsampleCombiner(nDivisions)
+
+  def result: IndexedSeq[Row] = _state.toRes
+
+  def seqOp(x: Any, y: Any) = {
```

Comment:
suggestion: If you don't want to check for null twice, you can make this seqOp `private`. That way only your method that has checked for null can call it.